README.md

```diff
@@ -38,41 +43,99 @@ on to create a full desktop environment.
+work for a long time hard. We want the latter.
+
+[KDE/kwin] and [GNOME/mutter] are great compositors offering brilliant
+experiences, but at ca 200k and 400k lines-of-code respectively and with heavy
+reliance on their respective eco-systems, they cannot be an elegant solution to
+the need for a frugal, independent, Wayland stacking compositor which is the
+space that labwc occupies.
+
+The Wayland community has achieved an amazing amount so far, and we want help
+solve the unsolved problems to make Wayland viable for more people.
+
+We think that standardisation and de-fragmentation is the route to greater
+Wayland adoption. Wanting to play our part in this, Labwc only understands
+[wayland-protocols] &amp; [wlr-protocols], and it cannot be controlled with
+dbus, sway/i3/custom-IPC or other technology.
 
-In order to avoid reinventing configuration and theme syntax, the [openbox] 3.6
-specification is used. This does not mean that labwc is an openbox clone but
-rather that configuration files will look and feel familiar.
+### 1.2 Why The Openbox Theme Specification?
 
-Labwc supports the following:
+In order to avoid reinventing configuration and theme syntaxes, the [openbox]
+3.6 specification is used. This does not mean that labwc is an openbox clone
+but rather that configuration files will look and feel familiar.
+
+Also, parsing GTK3+ and Qt themes for window decorations is very complicated,
+so using much simpler specs such as those used by openbox and xfwm makes sense
+for a compositor such as labwc, both in terms of implementation and for user
+modification.
+
+Openbox spec is somewhat of a stable standard considering how long it has
+remained unchanged for and how wide-spread its adoption is by lightweight
+distributions such as LXDE, LXQt, BunsenLabs, ArchLabs, Mabox and Raspian. Some
+widely used themes (for example Numix and Arc) have built-in support.
+
+We could have invented a whole new syntax, but that's not where we want to
+spend our effort.
+
+### 1.3 Very High Level Scope
+
+A lot of emphasis is put on code simplicity when considering features.
+
+The main development effort is focused on producing a solid foundation for a
+stacking compositor rather than adding configuration and theming options.
+
+See [scope] for full details on implemented features.
+
+High-level summary of items that Labwc supports:
 
 - [x] Config files (rc.xml, autostart, environment, menu.xml)
 - [x] Theme files and xbm icons
-- [x] Basic root-menu and client-menu
+- [x] Basic desktop and client menus
 - [x] HiDPI
 - [x] wlroots protocols such as `output-management`, `layer-shell` and
   `foreign-toplevel`
 - [x] Optionally xwayland
 
-See [scope] for full details on implemented features.
-
-## 1.1 Screenshot
+Items which are *not* intended to be implemented:
 
-<a href="https://i.imgur.com/vOelinT.png">
-  <img src="https://i.imgur.com/vOelinTl.png">
-</a>
+- Icons (except window buttons)
```

Comment:
This line isn't new, I know, but I've wondered whether it means that application icons within title bars are in or out of scope. It's something I miss from Openbox.

Reply:
Application icons are really tricky with wayland (coming from somebody who implemented application icon lookup for a panel). Wayland doesn't provide any notion of icons but has the app-id. By convention (!) the app-id is mapped to a `.desktop` file (e.g. sharing the same name) which then usually contains an icon name.

This may break for some xwayland applications, applications using CamelCase for the app-id but for their `.desktop` its all lowercase, filesystem namespaces (e.g. flatpack) which don't install their `.desktop` files in the compositor filesystem namespace and various other possible issues.

To sum up: figuring out the correct desktop icon is hard. We \*could\* implement it as we are depending on glib anyway but personally I don't think the complexity outweighs the benefits.

Reply:
Hmm, that's unfortunate but makes sense. I was hoping it worked like X11 where the application provides the image data.